### PR TITLE
STG94 Error Code For Copy Blob

### PR DESCRIPTION
**Pull request by ibrandes**

### CODE REVIEW

File: sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/policy/RequestRetryPolicy.java [hunk 1]
```diff
@@ -377,12 +378,27 @@ static ExceptionRetryStatus shouldErrorBeRetried(Throwable error, int attempt, i
...
 
-    static boolean shouldStatusCodeBeRetried(int statusCode, boolean isPrimary) {
+    static boolean shouldResponseBeRetried(int statusCode, boolean isPrimary, HttpResponse response) {
         /*
          * Retry the request if the server had an error (500), was unavailable (503), or requested a backoff (429),
          * or if the secondary was being tried and the resources didn't exist there (404). Only the secondary can retry
          * if the resource wasn't found as there may be a delay in replication from the primary.
          */
+        if (response != null) {
+            String headerValue = response.getHeaders().getValue(HttpHeaderName.fromString("x-ms-copy-source-error-code"));
```
Comment (alzimmermsft) [inline sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/policy/RequestRetryPolicy.java:388]: Make the `HttpHeaderName` a constant

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/implementation/models/BlobsGetAccountInfoHeaders.java [hunk 2]
```diff
@@ -30,12 +30,6 @@ public final class BlobsGetAccountInfoHeaders {
     @JsonProperty(value = "x-ms-account-kind")
     private AccountKind xMsAccountKind;
 
-    /*
-     * The x-ms-is-hns-enabled property.
-     */
-    @JsonProperty(value = "x-ms-is-hns-enabled")
```
[the remark this answers is not in the document]
Comment (ibrandes) [reply sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/implementation/models/BlobsGetAccountInfoHeaders.java:36]: Thanks for catching this, not sure why these got removed, they were from the Get Account Info on Container and Blob level feature.

File: sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/implementation/models/StorageError.java [hunk 3]
```diff
@@ -18,6 +18,25 @@ public final class StorageError {
     @JsonProperty(value = "Message")
     private String message;
 
+    /*
+<<<<<<< HEAD
```
Comment (ibrahimrabab) [inline sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/implementation/models/StorageError.java:22]: Might want to remove this

File: sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/implementation/models/StorageError.java [hunk 4]
```diff
@@ -47,6 +66,67 @@ public StorageError setMessage(String message) {
         return this;
     }
 
+    /**
+<<<<<<< HEAD
```
Comment (ibrahimrabab) [inline sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/implementation/models/StorageError.java:70]: This too

File: sdk/storage/azure-storage-file-share/src/test/java/com/azure/storage/file/share/FileAsyncApiTests.java [hunk 5]
```diff
@@ -778,6 +794,23 @@ public void startCopyError() {
             .verify(Duration.ofMinutes(1));
     }
 
+    @RequiredServiceVersion(clazz = BlobServiceVersion.class, min = "2024-08-04")
```
Comment (ibrahimrabab) [inline sdk/storage/azure-storage-file-share/src/test/java/com/azure/storage/file/share/FileAsyncApiTests.java:797]: Same here for ShareServiceVersion

File: sdk/storage/azure-storage-file-share/src/test/java/com/azure/storage/file/share/FileApiTests.java [hunk 6]
```diff
@@ -1316,6 +1332,21 @@ public void startCopyError() {
         FileShareTestHelper.assertExceptionStatusCodeAndMessage(e, 400, ShareErrorCode.INVALID_HEADER_VALUE);
     }
 
+    @RequiredServiceVersion(clazz = BlobServiceVersion.class, min = "2024-08-04")
```
Comment (ibrahimrabab) [inline sdk/storage/azure-storage-file-share/src/test/java/com/azure/storage/file/share/FileApiTests.java:1335]: Same here about ShareServiceVersion

**Review by alzimmermsft — APPROVED** [review]

**Pull request merged**